app/src/main/java/com/example/fond/adapters/RecipeSearchAdapter.java

```diff
@@ -56,15 +129,95 @@ public void onClick(View view) {
+
+        final List<Integer> queriedRecipes = new ArrayList<>();
+        query.findInBackground(new FindCallback<RecipeFavorites>() {
+            @Override
+            public void done(List<RecipeFavorites> recipeFavorites, ParseException e) {
+                if (recipeFavorites != null){
+                    for (RecipeFavorites recipeFavorite : recipeFavorites ) {
+                        Log.i(TAG, "Result: " + recipeFavorite.toString());
+                        ParseRecipe parseRecipe = (ParseRecipe) recipeFavorite.getParseObject("recipeId");
+                        queriedRecipes.add(parseRecipe.getRecipeId());
+                    }
+                }
+
+            }
+        });
+        return queriedRecipes;
+    }
+
+    private void checkRecipeFavoriteParse(final Recipe recipe) {
+        //Check if recipe added to parse
+        ParseQuery<ParseRecipe> query = ParseQuery.getQuery("ParseRecipe");
+        query.whereEqualTo("recipeId", recipe.getId());
+        query.getFirstInBackground(new GetCallback<ParseRecipe>() {
+            @Override
+            public void done(ParseRecipe object, ParseException e) {
+                if (object != null){
+                    Log.i(TAG, "Recipe Exists");
+                    RecipeFavorites newFavorite = new RecipeFavorites();
+                    newFavorite.put("userId", ParseUser.getCurrentUser());
+                    newFavorite.put("recipeId", object);
+
+                    newFavorite.saveInBackground(new SaveCallback() {
+                        @Override
+                        public void done(ParseException e) {
+                            Log.e(TAG, "Error: " + e );
+                        }
+                    });
+                }
+                else {
+                    Log.i(TAG, "Need to add recipe");
+                    final ParseRecipe newRecipe = new ParseRecipe();
+                    newRecipe.put("title", recipe.getTitle());
+                    newRecipe.put("imageUrl", recipe.getImage());
+                    newRecipe.put("summary", recipe.getSummary());
+                    newRecipe.put("recipeId", recipe.getId());
+
+                    newRecipe.saveInBackground(new SaveCallback() {
+                        @Override
+                        public void done(ParseException e) {
+                            if (e != null){
+                                Log.e(TAG, "Error: " + e );
+                            } else {
+                                RecipeFavorites newFavorite = new RecipeFavorites();
+                                newFavorite.put("userId", ParseUser.getCurrentUser());
+                                newFavorite.put("recipeId", newRecipe);
+
+                                newFavorite.saveInBackground(new SaveCallback() {
+                                    @Override
+                                    public void done(ParseException e) {
+                                        Log.e(TAG, "Error: " + e );
```

Comment:
I pretty sure this is a false positive error message to since it's being called when e is null. I'll go ahead and commit the change, test if everything works again, and approve the pull request.